ecs-agent/modeltransformer/transformer.go

```diff
@@ -0,0 +1,108 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package modeltransformer
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+)
+
+const (
+	modelTypeTask = "Task"
+)
+
+// Transformer stores transformation functions for all types of objects.
+// Transform<type> will execute a series of transformation functions to make it compatible with current agent version.
+// add<type>TransformationFunctions will add more <type> transformation functions to the transformation functions chain.
+// Add other transformation functions as needed. e.g. ContainerTransformationFunctions.
+// Add corresponding Transform<Type> and Add<Type>TransformationFunctions while adding other transformation functions.
+type Transformer struct {
+	taskTransformFunctions []*TransformFunc
+}
+
+// TransformFunc contains the threshold version string for transformation function and the transformationFunction itself.
+// During upgrade, all models from versions below threshold version should execute the transformation function.
+type TransformFunc struct {
+	version  string
+	function interface{}
+}
+
+func NewTransFormer() *Transformer {
+	t := &Transformer{}
+	return t
+}
+
+// GetNumberOfTransformationFunctions returns the number of transformation functions given a model type
+func (t *Transformer) GetNumberOfTransformationFunctions(modelType string) int {
+	switch modelType {
+	case modelTypeTask:
+		return len(t.taskTransformFunctions)
+	default:
+		return 0
+	}
+}
+
+// TransformTask executes the transformation functions when version associated with model in boltdb is below the
```

Comment:
Seems like this comment is incomplete.

Reply:
oops it seems to be a mis-edit. I will fix it!